src/tries/serializeTrie.js

```diff
@@ -42,14 +42,15 @@ function lineToString(line, i, arr) {
         const prevLine = arr[i - 1];
 
         indexOfDifference = findIndexOfDifference(line, prevLine);
+
         if (indexOfDifference === -1) {
             // Identical lines
             return "";
         }
         if (indexOfDifference === 0) {
             // line and prevLine are completely different
             separatorFromPrev = SEPARATORS.RESET;
-        } else if (prevLine.length === line.length && indexOfDifference === line.length - 1) {
+        } else if (prevLine.length <= line.length && indexOfDifference === prevLine.length - 1) {
```

Comment:
I think the whole `length` check is not necessary or too strict. We should just compare `indexOfDifference === prevLine.length - 1)